dockerfiles/Dockerfile

```diff
@@ -76,6 +76,7 @@ RUN cmake -DFLB_RELEASE=On \
     -DFLB_NIGHTLY_BUILD="$FLB_NIGHTLY_BUILD" \
     -DFLB_LOG_NO_CONTROL_CHARS=On \
     -DFLB_CHUNK_TRACE="$FLB_CHUNK_TRACE" \
+    -DFLB_REGEX_RUBY=On \
```

Comment:
This also needs to flow into the package builds and there I would say it needs to be configurable via an ARG:
- ARG with default: 
https://github.com/fluent/fluent-bit/blob/2b77938867723a8d5eccab942864c8327410d1a2/packaging/distros/ubuntu/Dockerfile#L147

- CMake config: 
https://github.com/fluent/fluent-bit/blob/2b77938867723a8d5eccab942864c8327410d1a2/packaging/distros/ubuntu/Dockerfile#L160

Once done we also need to ensure this builds for all targets - see the template above to run a local test first then we can label with `ok-package-test`.